formats/tabular-to-arrow-plugin/src/polus/plugins/formats/tabular_to_arrow/__main__.py

```diff
@@ -0,0 +1,98 @@
+"""Tabular to Arrow."""
+import json
+import logging
+import pathlib
+from concurrent.futures import ProcessPoolExecutor, as_completed
+from multiprocessing import cpu_count
+from typing import Any, Optional
+
+import filepattern as fp
+import typer
+from tqdm import tqdm
+
+from polus.plugins.formats.tabular_to_arrow import tabular_arrow_converter as tb
+
+app = typer.Typer()
+# Set number of processors for scalability
+max_workers = max(1, cpu_count() // 2)
+
+# Initialize the logger
+logging.basicConfig(
+    format="%(asctime)s - %(name)-8s - %(levelname)-8s - %(message)s",
+    datefmt="%d-%b-%y %H:%M:%S",
+)
+logger = logging.getLogger("polus.plugins.formats.tabular_to_arrow")
+
+FILE_EXT = ".arrow"
```

Comment:
The input files should always be `.arrow` files. The output needs to be taken from the environment variable `POLUS_TAB_EXT`, and then overruled based on the CLI arguments.